### PR TITLE
ansible: Add new aix7.1 machines

### PR DESCRIPTION
**Pull request by AshCripps**
Our current AIX7.1 machines are on a service contract which expires at the end of the year. This PR will add 2 new test machines and a release machine to replace. im also going to try and add some ansible tweaks to remove some of the metric &%^*load of manual steps we have for AIX.

### CODE REVIEW

**Comment by AshCripps** [comment]
I have added a new partial to bootstrap AIX machines to the same size file system (removes some of the manual steps) this all works bar the /aha fs which didnt set itself up properly so will need to look into that. I also plan to add the ccache install steps to ansible to remove some more manual steps (untill we can download a package)

would appreciate some reviews of whats there so far.

cc: @richardlau @aixtools @mhdawson

File: ansible/roles/bootstrap/tasks/partials/aix.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+
+#
+# Setup the filesystem to the correct sizes on AIX
+#
+
+- name: Set size of / to 1G
+  aix_filesystem:
```
Comment (richardlau) [inline ansible/roles/bootstrap/tasks/partials/aix.yml:8]: Is this part of the standard ansible install? Do we need to update instructions to use this?

Comment (AshCripps) [reply ansible/roles/bootstrap/tasks/partials/aix.yml:8]: it should be part of standard past a certain version 2.8+ im on 2.8.5 and didnt need to install anything extra

**Comment by AshCripps** [comment]
I got /aha to mount (I think) @richardlau can you add the ips to the firewall so I can do some testing please?

**Comment by richardlau** [comment]
> 
> 
> I got /aha to mount (I think) @richardlau can you add the ips to the firewall so I can do some testing please?
> EDIT: one may already be there but under the wrong name

Done (and updated the names to match the IP addresses listed in this PR).

**Comment by AshCripps** [comment]
Ive added ansible steps for the building of ccache its a little long winded but does all that the manual steps did before and cleans up after itself. im going to move this out of draft - what remains is a release machine and testing/connecting to jenkins

File: ansible/roles/jenkins-worker/tasks/main.yml [hunk 2]
```diff
@@ -77,6 +77,76 @@
   when: "'scaleway-ubuntu1804-armv7l' in inventory_hostname"
   include: "{{ role_path }}/tasks/partials/scaleway-armv7.yml"
 
+- name: aix install ccache - check for executable
```
Comment (richardlau) [inline ansible/roles/jenkins-worker/tasks/main.yml:80]: Is it possible to extract some commonality with what we do for smartos/s390x?
[`ansible/roles/baselayout/tasks/ccache.yml`](https://github.com/nodejs/build/blob/5e9585a7014ac631f2c5216dd5422e666c7636dc/ansible/roles/baselayout/tasks/ccache.yml)
https://github.com/nodejs/build/blob/06c83e72a359ff8b24e5b04bf071656f6325c2e3/ansible/roles/baselayout/vars/main.yml#L7-L10
https://github.com/nodejs/build/blob/fdd596c436764ca0d55ca210b1d16aa619e6a146/ansible/roles/baselayout/tasks/main.yml#L59-L62
https://github.com/nodejs/build/blob/fdd596c436764ca0d55ca210b1d16aa619e6a146/ansible/roles/baselayout/tasks/main.yml#L196-L206

Comment (AshCripps) [reply ansible/roles/jenkins-worker/tasks/main.yml:80]: not 100% sure we use a different version everywhere (that seems real bad practise) and also AIX installs the ccache elsewhere. maybe that could be part of another PR to clean up how we install certain not package manager binaries?

Comment (AshCripps) [reply ansible/roles/jenkins-worker/tasks/main.yml:80]: actually looking at `ccache.yml` its probs worth at least moving what I have to there

Comment (AshCripps) [reply ansible/roles/jenkins-worker/tasks/main.yml]: reabsed and moved to `ccache.yml` and it works

File: ansible/roles/baselayout/tasks/ccache.yml [hunk 3]
```diff
@@ -5,25 +5,40 @@
...
 
 - name: "ccache : fetch samba ccache HTML page"
+  when: not os|startswith("aix")
   # This depends on ansible being able to run curl locally, YMMV, if it doesn't
   # work, try changing the local_action to a remote one:
   #  raw: curl -sL https://www.samba.org/ftp/ccache/
   local_action: shell curl -sL https://www.samba.org/ftp/ccache/ | sed -n -e 's/.*"ccache-\([0-9.]*\).tar.gz".*/\1/p' | tail -1
   register: ccache_html_content
 
+- name: "ccache : fetch tarbal - aix"
```
Comment (richardlau) [inline ansible/roles/baselayout/tasks/ccache.yml:15]:
```suggestion
- name: "ccache : fetch tarball - aix"
```

**Review by richardlau — APPROVED** [review]
LGTM. I think we can simplify `ccache.yml` but let's leave that for a separate PR.

**Comment by AshCripps** [comment]
https://ci.nodejs.org/job/node-test-commit-aix-ashcripps/nodes=test-ibm-aix71-ppc64_be-3/buildTimeTrend +
https://ci.nodejs.org/job/node-test-commit-aix-ashcripps/nodes=test-ibm-aix71-ppc64_be-4/buildTimeTrend

Build times look good so im going to add the test machines to the regular job

**Comment by aixtools** [comment]
> 
> 
> I have added a new partial to bootstrap AIX machines to the same size file system (removes some of the manual steps) this all works bar the /aha fs which didnt set itself up properly so will need to look into that. I also plan to add the ccache install steps to ansible to remove some more manual steps (untill we can download a package)
> 
> would appreciate some reviews of whats there so far.
> 
> cc: @richardlau @aixtools @mhdawson

My goal is to have at least two partitions up and running in OSU environment tomorrow.

I'll also create an id for you (@AshCripps ) or someone else. Just need a public key to add to your account after I make it. This is on my admin machine  acting as `Control Node` - so you can locally test ansible (as I am doing right now also for AdoptOpenJDK).

As I said - two systems - one that has yum.sh, and all the 'extras' that entails, and one with my packaging of Python (v3.6.12) which also supports Ansible. I'll add a comment here when they are up.

As Ansible is `new` to me, I am writing up a primer `Ansible on AIX` - largely to train myself, but I hope it may be useful to others - eventually. Just a simple wiki page: http://www.aixtools.net/index.php/ansible - feedback is always welcome!

**Comment by aixtools** [comment]
I guess I should have asked - these two systems - where are they located - the generic answer is where do I look to see any (AIX) systems mapping between nodejs names and DNS names.

My comment has to do with what I am doing for #2424 and the creation of two new AIX 7.2 endpoints.

re: ansible - I am working on a new playbook for AdoptOpenJDK - and much of what I am doing there will likely be applicable here.

**Comment by AshCripps** [comment]
@aixtools these machines are in IBM cloud and are completly seperate from OSUOSL

**Comment by aixtools** [comment]
OK. I'll proceed using #2424

**Comment by mhdawson** [comment]
@aixtools thanks, looking forward to getting those AIX 7.2 machines from OSU into the CI.

**Review by mhdawson — APPROVED** [review]
LGTM

**Comment by AshCripps** [comment]
ive added the release machine and done the manual steps to connect it. @richardlau can you add it to the release firewall so I can run a test build before merging the PR?

**Comment by AshCripps** [comment]
https://ci-release.nodejs.org/job/iojs+release/nodes=aix71-ppc64/6380/ passed

**Pull request merged**